tests/unit/template_cassdc_test.go

```diff
@@ -52,6 +52,8 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 
 			Expect(renderTemplate(options)).To(Succeed())
 
+			validateRequiredLabels(cassdc.Labels)
```

Comment:
Instead of modifying each test to validate the labels, wouldn't it make sense to render the whole Helm chart and check that each rendered yaml has labels attached to it? So basically dry-run the chart install with all the options enabled (ingress, stargate etc)